### ISSUE TITLE
bump kserve

### ISSUE THREAD

**Issue by mmoya91** [kserve/kserve#3327]
/kind bug

I noticed kserve v0.11.2 is out which should include a bumped version of ray
https://pypi.org/project/kserve/

but kserve itself was never bumped/published. Looks like it's still on v0.11.1:
https://github.com/kserve/kserve/blob/master/python/kserve/pyproject.toml#L3

**Comment by terrytangyuan** [comment]
Looks like an oversight. We'll be releasing 0.12 soon so maybe upgrade to use that instead once it's available?

**Comment by sivanantha321** [comment]
@mmoya91 Kserve is [bumped in branch release-0.11](https://github.com/kserve/kserve/blob/b688db6be4e640cfbc89030dd802428652da58c0/python/kserve/pyproject.toml#L3). Upgrading ray serve is not part of 0.11.2 release.

**Comment by mmoya91** [comment]
@sivanantha321 I'm running into the following error whenever I try installing kserve `0.11.2` via poetry
```
 Unable to find installation candidates for ray (2.4.0)

  at /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/installation/chooser.py:109 in choose_for
      105│ 
      106│             links.append(link)
      107│ 
      108│         if not links:
    → 109│             raise RuntimeError(f"Unable to find installation candidates for {package}")
      110│ 
      111│         # Get the best link
      112│         chosen = max(links, key=lambda link: self._sort_key(package, link))
```
I had to install `0.12.0rc0` as a workaround

**Comment by sivanantha321** [comment]
Please, Use patched 0.12.1 version
